sdks/python/apache_beam/metrics/internal.py

```diff
@@ -0,0 +1,116 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""
+Internal classes for Metrics API.
+
+The classes in this file keep shared state, and organize metrics information.
+
+Available classes:
+- MetricsEnvironment - Keeps track of MetricsContainer and other metrics
+    information for every single execution working thread.
+- MetricsContainer - Holds the metrics of a single step and a single
+    unit-of-commit (bundle).
+"""
+from collections import defaultdict
+import threading
+
+from apache_beam.metrics.base import MetricKey, MetricUpdates
+from apache_beam.metrics.cells import CounterCell, DistributionCell
+
+
+class MetricsEnvironment(object):
+  """ Holds the MetricsContainer for every thread and other metric information.
+
+  This class is not meant to be instantiated, instead being used as to keep
+  track of global state.
+  """
+  METRICS_SUPPORTED = False
+  _METRICS_SUPPORTED_LOCK = threading.Lock()
+
+  PER_THREAD = threading.local()
+
+  @classmethod
+  def set_metrics_supported(cls, supported):
+    with cls._METRICS_SUPPORTED_LOCK:
+      cls.METRICS_SUPPORTED = supported
+
+  @classmethod
+  def current_container(cls):
+    try:
+      return cls.PER_THREAD.container
+    except AttributeError:
+      return None
+
+  @classmethod
+  def set_current_container(cls, container):
+    cls.PER_THREAD.container = container
+
+  @classmethod
+  def unset_current_container(cls):
+    del cls.PER_THREAD.container
+
+
+class MetricsContainer(object):
+  """ Holds the metrics of a single step and a single bundle.
+  """
+  def __init__(self, step_name):
+    # Import here to avoid the circular dependency
+    self.step_name = step_name
+    self.counters = defaultdict(lambda: CounterCell())
+    self.distributions = defaultdict(lambda: DistributionCell())
+
+  def get_Counter(self, metric_name):
```

Comment:
Is this naming convention standard? Should it be either `snake_case` or `camelCase`? This seems to mix the two.